var/spack/repos/builtin/packages/duckdb/package.py

```diff
@@ -16,8 +16,106 @@ class Duckdb(CMakePackage):
 
     license("MIT")
     maintainers("glentner")
+    maintainers("teaguesterling")
 
     version("master", branch="master")
+    version("0.10.0", sha256="5a925b8607d00a97c1a3ffe6df05c0a62a4df063abd022ada82ac1e917792013")
     version("0.9.2", sha256="afff7bd925a98dc2af4039b8ab2159b0705cbf5e0ee05d97f7bb8dce5f880dc2")
+    version("0.9.1", sha256="37a43188d9354ce3ca101b2b118d867f5f76d04c3b83c09d86fd7508351a631b")
+    version("0.9.0", sha256="3dbf3326a831bf0797591572440e81a3d6d668f8e33a25ce04efae19afc3a23d")
     version("0.8.1", sha256="a0674f7e320dc7ebcf51990d7fc1c0e7f7b2c335c08f5953702b5285e6c30694")
+    version("0.8.0", sha256="df3b8e0b72bce38914f0fb1cd02235d8b616df9209beb14beb06bfbcaaf2e97f")
     version("0.7.1", sha256="67f840f861e5ffbe137d65a8543642d016f900b89dd035492d562ad11acf0e1e")
+
+    depends_on("python@3:")
+    depends_on("cmake")
+    depends_on("gmake")
+    depends_on("ninja", when="+ninjabuild")
+    depends_on("openssl", when="+httpfs")
+    depends_on("icu4c", when="~icu")
+
+    # Build Options
+    variant("autocomplete", default=True, description="Include autocomplete for CLI in build")
+    variant("cli", default=True, description="Compile with command line client")
+    variant("icu", default=False, description="Compile with bundled ICU library")
+    variant("ninjabuild", default=True, description="Use GEN=ninja to build")
```

Comment:
I'm a bit puzzled by the way this is packaged. We use `gmake` to drive `cmake` to generate `ninja` builds :thinking: I get that this is how upstream advise to build from sources... but is there a real drawback in using CMake directly?[^1]

Note that `patch` can be called independently from the base class being selected.

[^1]: Somehow I'm not persuaded by `It is not advised to directly call CMake, as the Makefile sets certain variables that are crucial to properly building the package.` Wondering if you have a bit more info on that.

Reply:
It seems that there is some logic in the Makefile that handles multiple cmake builds for supporting extensions, but the default behavior doesn't trigger those additional targets. That said, the Makefile appears to be fairly actively developed. I initially worried that it would be difficult to keep the cmake variable setting in sync, especially around extension handling.